src/resolvers/path.cr

```diff
@@ -41,16 +41,29 @@ module Shards
     end
 
     private def check_install_path_target
-      begin
-        real_install_path = File.real_path(install_path)
-      rescue errno : Errno
-        if errno.errno == Errno::ENOENT
-          return false
-        else
-          raise errno
+      {% if compare_versions(Crystal::VERSION, "0.34.0-0") > 0 %}
+        begin
+          real_install_path = File.real_path(install_path)
+        rescue ex : File::Error
```

Comment:
This should be `File::NotFoundError` instead, and no need to check the `os_error`